ocs_ci/utility/framework/fusion_fdf_init.py

```diff
@@ -0,0 +1,237 @@
+
+        Args:
+            args (list): List of args passed to CLI
+
+        Returns:
+            list: List of parsed args
+
+        """
+        logger.info("Parsing arguments")
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--cluster-name", help="Name of the OCP cluster")
+        parser.add_argument("--cluster-path", help="OCP cluster directory")
+        parser.add_argument(
+            "--conf",
+            action="append",
+            default=[],
+            help="Path to config file. Repeatable.",
+        )
+        parsed_args, _ = parser.parse_known_args(args)
+
+        return parsed_args
+
+    def init_logging(self) -> None:
+        """
+        Initialize the logging config.
+        """
+        log_dir = config.RUN["log_dir"]
+        log_level = config.RUN.get("log_level", "INFO")
+        log_name = f"{self.log_basename}_{self.run_id}"
+        log_formatter = logging.Formatter(constants.LOG_FORMAT)
+        root_logger = logging.getLogger()
+        root_logger.setLevel(log_level)
+
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+
+        log_file = os.path.join(log_dir, f"{log_name}.log")
+
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(log_formatter)
+        file_handler.setLevel(log_level)
+        root_logger.addHandler(file_handler)
+
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(log_formatter)
+        console_handler.setLevel(log_level)
+        root_logger.addHandler(console_handler)
+
+        logger.info("Logging initialized")
+        logger.info(f"Log file configured: {log_file}")
+
+    def set_cluster_connection(self) -> None:
+        """
+        Setup cluster connection.
+        """
+        logger.info("Setting kubeconfig")
+        config.RUN["kubeconfig"] = os.path.join(
+            config.ENV_DATA["cluster_path"], config.RUN["kubeconfig_location"]
+        )
+        setup_bin_dir()
+        set_kubeconfig(config.RUN["kubeconfig"])
+
+
+def generate_run_id() -> int:
+    """
+    Generate run_id for the deployment.
+
+    Returns:
+        int: Unique identifier for the run
+
+    """
+    logger.debug("Generating run_id from timestamp")
+    run_id = int(time.time())
+    config.RUN["run_id"] = run_id
+    return run_id
+
+
+def set_kubeconfig(kubeconfig_path: str):
+    """
+    Export environment variable KUBECONFIG for future calls of OC commands
+    or other API calls
+
+    Args:
+        kubeconfig_path (str): path to kubeconfig file to be exported
+
+    Raises:
+        ClusterNotAccessibleError: if the cluster is inaccessible
+
+    """
+    logger.info("Testing access to cluster with %s", kubeconfig_path)
+    if not os.path.isfile(kubeconfig_path):
+        raise ClusterNotAccessibleError(
+            "The kubeconfig file %s doesn't exist!", kubeconfig_path
+        )
+    os.environ["KUBECONFIG"] = kubeconfig_path
```

Comment:
If we can avoid the KUBECONFIG in env variable the better, as it's a main problem in running multiple threads in multicluster scenario as ENV is used across whole framework

Reply:
I've changed this function to just check cluster access with the provided kubeconfig, without setting the env var as well as updated all `oc` commands to pass the kubeconfig directly.